tests/shell/tests/01-run.py

```diff
@@ -63,6 +63,9 @@ def check_cmd(child, cmd, expected):
 
 
 def testfunc(child):
+    # wait for prompt
+    child.expect_exact('test_shell.\r\n>')
```

Comment:
Having to wait for `test_shell` as well requires resetting the board/application after the terminal is opened, I'm trying to get rid of that dependency in tests since a lot of boards don't have `make reset` support.

```suggestion
    child.expect_exact('>')
```

Reply:
Actually the prompt can still be missed.

Reply:
Although I don't like using a shell function to test the shell, maybe we should just enable `test_utils_interactive_sync` it would get rid of the issues with synchronization for good, applying the following patch works:

https://gist.github.com/fjmolinas/57b335882b8802317e7aae19096c5789

Reply:
```
    child.sendline('')
    child.expect('>')
```

would do if `make reset` is not supported, right?

Reply:
Or re-use "CONTROL_C". please take a look at the fixup.

Reply:
> maybe we should just enable `test_utils_interactive_sync` it would get rid of the issues with synchronization for good

I'd rather fix the test script, if possible.

Reply:
Works as expected, nice one!

`RESET= BOARD=iotlab-m3 make -C tests/shell/ flash test`

<details>
> 
> start_test
start_test
[TEST_START]
> end_test
end_test
[TEST_END]
> 

123456789012345678901234567890123456789012345678901234567890

> 
> 123456789012345678901234567890123456789012345678901234567890
shell: command not found: 123456789012345678901234567890123456789012345678901234567890
> unknown_command
unknown_command
shell: command not found: unknown_command
> help
help
Command              Description
---------------------------------------
start_test           starts a test
end_test             ends a test
echo                 prints the input command
reboot               Reboot the node
ps                   Prints information about running threads.
app_metadata         Returns application metadata
> echo a string
echo a string
"echo""a""string"
> ps
ps
        pid | state    Q | pri 
          1 | pending  Q |  15
          2 | running  Q |   7
> garbage1234
help
garbage1234
> 
> help
Command              Description
---------------------------------------
start_test           starts a test
end_test             ends a test
echo                 prints the input command
reboot               Reboot the node
ps                   Prints information about running threads.
app_metadata         Returns application metadata
> reboot
reboot
�main(): This is RIOT! (Version: 2020.01-devel-1386-ge2ad4-pr-12961)
test_shell.
> 
make: Leaving directory '/home/francisco/workspace/RIOT-test/tests/shell'
</details>